.github/workflows/build-deploy-api-v2.yaml

```diff
@@ -0,0 +1,119 @@
+          cmd: |
+            'version'
+            
+      # Login to BC Gov Docker Image Repository
+      - name: Login to Openshift Docker
+        run : |
+          docker login ${{ secrets.OPENSHIFT4_PUBLIC_IMAGE_REPOSITORY_BASE_URL }} -u ${{ secrets.OPENSHIFT4_TOOLS_SA_USERNAME }} -p ${{ secrets.OPENSHIFT4_TOOLS_SA_PASSWORD }}
+          
+      # Setup environment variables
+      - name: Setup environment
+        run: |
+          make api-setup-env
+      
+      # Build the DPIA API Application Image via docker-compose
+      - name: Build the Docker image
+        run: |
+          docker-compose build dpia-api
+      
+      # TROUBLESHOOTING - view images present in github action run
+      - name: Docker List Images
+        run: |
+          docker image ls 
+      
+      # Tag Docker Image built with latest
+      - name: Docker Tag Version
+        run: |
+          docker tag cirmo-dpia_dpia-api ${{ secrets.OPENSHIFT4_API_PUBLIC_IMAGE_REPO_DEV }}:latest
+
+      # Push docker image to BC Gov Repository  
+      - name: Docker Push Version
+        run: |
+          docker push ${{ secrets.OPENSHIFT4_API_PUBLIC_IMAGE_REPO_DEV }}:latest
+          
+  DEV-deploy:
+    runs-on: ubuntu-latest
+    environment: API-Dev
+    needs: [ DPIA-API-Build-Tag-Push ]
+    steps:
+    
+      # Checkout Repository
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+      
+      # Login to BC Gov Openshift (DPIA DEV Namespace)
+      - name: Cluster Login -> DEV Namespace
+        uses: redhat-developer/openshift-actions@v1.1
+        with:
+          openshift_server_url: ${{ secrets.OPENSHIFT4_SERVER_URL }}
+          parameters: '{"apitoken": "${{ secrets.OPENSHIFT4_SA_PASSWORD }}"}'
+          cmd: |
+            'version'
+
+      # Use deploy helper script to process and apply DPIA API Application template into DEV namespace
+      - name: Process and Apply API DeploymentConfig, Service, Route templates
+        env:
+          NAMESPACE: "${{ secrets.OPENSHIFT_DEV_NAMESPACE }}"
+          APPLICATION_NAME: "${{ secrets.APPLICATION_NAME }}"
+          LICENSE_PLATE: "${{ secrets.LICENSE_PLATE }}"
+          ENVIRONMENT: "${{ secrets.ENVIRONMENT }}"
+          DC_TEMPLATE: "api-dc-template-v2.yaml"
+          SERVICE_TEMPLATE: "api-service-template-v2.yaml"
+          ROUTE_TEMPLATE: "api-route-template-v2.yaml"
+          DIR: "api"
+          IMAGESTREAM: "api-dev"
+        run: |
+          ./.github/helpers/deploy.sh 
+    
+  DEV-health-check:
+    runs-on: ubuntu-latest
+    environment:
+      name: API-Dev
+      url: https://dev.pia.gov.bc.ca/api/api-docs
+    needs: [ DEV-deploy ]
+    steps:
+    
+      # Checkout Repository
+      - uses: actions/checkout@v2
+    
+      # Login to BC Gov Openshift (DPIA DEV Namespace)
+      - name: Cluster Login -> DEV Namespace
+        uses: redhat-developer/openshift-actions@v1.1
+        with:
+          openshift_server_url: ${{ secrets.OPENSHIFT4_SERVER_URL }}
```

Comment:
Ah there's a lot of things here I don't think are secrets. Possibly you are just using it as a general env var store.

Reply:
that was the idea but I'm trying not to expose anything internal to openshift in our repo.